scipy/linalg/basic.py

```diff
@@ -814,40 +836,121 @@ def lstsq(a, b, cond=None, overwrite_a=False, overwrite_b=False,
         nrhs = 1
     if m != b1.shape[0]:
         raise ValueError('incompatible dimensions')
-    gelss, = get_lapack_funcs(('gelss',), (a1, b1))
-    if n > m:
+
+    if lapack_driver not in ('gelsd', 'gelsy', 'gelss'):
+        raise ValueError('LAPACK driver "%s"is not found' % lapack_driver)
+
+    lapack_func, lapack_func_lwork_query = get_lapack_funcs((lapack_driver,
+                                        '%s_lwork' % lapack_driver), (a1, b1))
+    # the dtype isn't necessarily the dtype needed for lapack.  It's the 
+    # result of np.find_common_type on the array inputs. 
```

Comment:
Then this would be best fixed in `blas.find_best_blas_type` rather than adding workarounds